File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    int lengthOfListNode (ListNode* head) {
        int cnt = 0;

        while(head) {
            cnt++;
            head = head->next;
        }

        return cnt;
    }
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int length = lengthOfListNode(head);
        int nth = length - n;

        if (head == nullptr) return head;

        if (nth == 0 && head) return head->next;

        int cnt = 0;
        ListNode* temp = head;

        while (temp) {
            if (cnt == nth - 1) {
                temp->next = temp->next->next;
            }
            cnt++;
            temp = temp->next;
        }

        return head;

    }
};

File: Data Structures & Algorithms/remove-linked-list-elements/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dmp = ListNode(0, head)
        tmp = dmp
        while tmp.next:
            if tmp.next.val == val:
                tmp.next = tmp.next.next
            else:
                tmp = tmp.next
        return dmp.next